### PR TITLE
Bump Ariadne to 0.45.0

### PR DESCRIPTION
**Pull request by khatchad**
## Summary

- Bump `com.ibm.wala.cast.python.ml` from `0.44.0` to `0.45.0` in `hybridize.target`.
- Add `<classifier>fat</classifier>` to the Ariadne dependency **as an interim measure**. Per wala/ML#418 and ponder-lab/ML#298, the slim default JAR is the intended consumption mode for library consumers like Hybridize, but slim consumption is currently blocked—see Why Not Slim below.
- Sweep 42 `testHasLikelyTensorParameter*` expectations to Ariadne 0.45.0's typed dim sentinels (`DynamicDim`, `RaggedDim`) and `RaggedRange` precision win.

Closes #559.

## Why Not Slim

ponder-lab/ML#298 stated the slim default JAR is the intended consumption mode: "library consumers like Hybridize now get exactly what [wala/ML#418] asked for: a thin-jar published artifact with proper Maven dependency resolution". Tried on the parallel branch [`bump-ariadne-0.45.0-slim`](https://github.com/ponder-lab/Hybridize-Functions-Refactoring/tree/bump-ariadne-0.45.0-slim) (`includeDependencyDepth="infinite"`, no classifier), result:

```
[ERROR] Artifact org.python:jython3:0.0.1-SNAPSHOT ... could not be wrapped
as a bundle: The default package '.' is not permitted by the Import-Package
syntax.
```

`jython3` ships `ProxyDeserialization.class` at the default (unnamed) package level. Tycho 5+'s stricter bnd auto-wrap rejects this—same root cause documented in wala/ML#418 but a form wala/ML#418's fix doesn't cover. wala/ML#418's narrow patch (shade-exclude `ProxyDeserialization.class` from the fat JAR) keeps the fat path working but can't apply to the slim path: no shading means `jython3` flows through transitively with the bad class intact.

Tracking issue filed at wala/ML#557 (proposed fix: shade-exclude `ProxyDeserialization.class` at the `jython3` build step itself, mirroring the wala/ML#418 pattern one layer down).

**Once wala/ML#557 is resolved, this PR should be reworked** to drop the classifier and switch to `includeDependencyDepth="infinite"`.

## Test Expectation Sweep

`./mvnw verify` initially reported **42 failures** across `testHasLikelyTensorParameter*`. All were expectation drift, not regressions—Ariadne 0.45.0's typed-sentinel rollout (wala/ML#545 `DynamicDim`, ponder-lab/ML#320 `RaggedDim`) and one precision win (ponder-lab/ML#332 `RaggedRange`). Six unique diff patterns observed:

| Expected (0.44.0) | Actual (0.45.0) | Root Cause |
|---|---|---|
| `[null, D:Constant,32]` | `[D:Dynamic,null, D:Constant,32]` | wala/ML#545—`keras.Input` batch dim materialized as `DynamicDim` instead of raw `null` |
| `[D:Constant,4, null]` | `[D:Constant,4, D:Ragged,null]` | wala/ML#449 and ponder-lab/ML#320—ragged row dim materialized as `RaggedDim` instead of raw `null` |
| `[D:Constant,5, null]` | `[D:Constant,5, D:Ragged,null]` | same as above |
| `[D:Constant,3, null, null]` | `[D:Constant,3, D:Ragged,null, D:Ragged,null]` | same—applies to multi-axis ragged tensors |
| `[D:Constant,4, null, null, null]` | `[D:Constant,4, D:Ragged,null, D:Ragged,null, D:Dynamic,null]` | mixed—ragged dims + dynamic batch combined |
| `[D:Constant,1, null]` | `[D:Constant,1, D:Constant,5]` | ponder-lab/ML#332—`RaggedRange` precision improvement; `null` is now resolved to concrete `5` |

Confirmed via Python runtime: `keras.Input(shape=(32,))` followed by addition produces shape `(None, 32)`—rank 2—and Ariadne 0.45.0 returns rank 2 (`[D:Dynamic, D:Constant,32]`). `D:Dynamic,null` is the toString of a single `DynamicDim` (type=`Dynamic`, payload value=`null` since the type parameter is `Void`).

For the `ragged.range` case (sixth row above), both the parameter `TensorType` and the inferred signature `TensorType` shifted: signature dims tighten from `SymbolicDim("?")` to `NumericDim(5)` in lockstep.

### Fix Applied

Mechanical sweep on the Hybridize side. For each failing test, the `Arrays.asList(null, ...)` or `Arrays.asList(..., null, ...)` argument in the expected `TensorType` had `null` replaced by the appropriate typed sentinel:

- Batch dim (leading position on a non-ragged tensor) → `DynamicDim.INSTANCE`.
- Ragged row dim (interior position on a `RaggedTensor`) → `RaggedDim.INSTANCE`.
- Flat values dim (trailing position on a multi-axis `RaggedTensor`) → `DynamicDim.INSTANCE`.
- `RaggedRange` parameter and signature: `null` and `SymbolicDim("?")` → `NumericDim(5)`.

7 `keras.Input` sites + ~35 ragged-tensor sites updated. `DynamicDim` and `RaggedDim` imports added. The now-resolved `wala/ML#544` typed-sentinel TODO was dropped from `testHasLikelyTensorParameter59`'s docstring; the `Hybridize#524` `RaggedTensorSpec` emission TODO stays.

## Other Notes

- Without the `fat` classifier, Tycho target-platform resolution fails with `Missing requirement: ... requires 'java.package; com.google.common.collect 0.0.0'`—the slim default JAR doesn't bundle Guava (it's a transitive POM dep), and `includeDependencyDepth="none"` doesn't follow the POM.
- Build uses JDK 25 (matches CI's `actions/setup-java` config).
- Test count: 492 (compared to 471 baseline on the 0.44.0 bump at #493)—21 new tests added in the interim.

## Test Plan

- [x] Test expectations regenerated for the 42 affected `testHasLikelyTensorParameter*` cases (see table above).
- [ ] CI passes on the expectation-regenerated branch.
- [x] Hybridize consumer migration is tracked: #561 covers explicit `instanceof DynamicDim` and `instanceof RaggedDim` branching in `Function.inferSpec`, satisfying the "or a follow-up issue tracks that migration" half of #559's acceptance.

### CODE REVIEW

File: edu.cuny.hunter.hybridize.tests/test cases/edu/cuny/hunter/hybridize/tests/HybridizeFunctionRefactoringTest.java [hunk 1]
```diff
@@ -3063,18 +3065,17 @@ private void testHasLikelyTensorParameterHelperMultiFunction(String targetFuncti
...
-	 * a {@code TensorSpec}-shaped signature that admits the runtime call but loses the raggedness signal.
-	 * TODO(https://github.com/ponder-lab/Hybridize-Functions-Refactoring/issues/524): flip from a {@code TensorSpec} with wildcards to a
-	 * {@code RaggedTensorSpec} once the consumer-side emission lands.
+	 * Test for #2 for TF API `RaggedTensor.from_nested_row_splits`. Canonical fixture for the ragged-tensor precision gap. Runtime:
+	 * {@code tf.RaggedTensor.from_nested_row_splits(values, splits)} dtype=int32, shape=(3, None, None). Ariadne emits an INT32
+	 * {@link TensorType} with three dims: a known constant 3 followed by two {@link RaggedDim} raggedness markers (per wala/ML#544 /
+	 * ponder-lab/ML#320, shipped in Ariadne 0.45.0). The inference algorithm collapses ragged markers to {@code SymbolicDim("?")} wildcards
+	 * in the inferred signature, producing a {@code TensorSpec}-shaped signature that admits the runtime call but loses the raggedness
+	 * signal. TODO(https://github.com/ponder-lab/Hybridize-Functions-Refactoring/issues/524): flip from a {@code TensorSpec} with wildcards
```
[the remark this answers is not in the document]
Comment (khatchad) [reply edu.cuny.hunter.hybridize.tests/test cases/edu/cuny/hunter/hybridize/tests/HybridizeFunctionRefactoringTest.java:3073]: > The linked issue's acceptance criteria require either migrating at least one `TensorType.getDims()` consumer to type-discriminate `DynamicDim`/`RaggedDim` or linking a follow-up; otherwise `Closes #559` would leave that acceptance item incomplete.

Filed #561 to track the `Function.inferSpec` consumer migration. PR body's test plan now references it; the "or a follow-up issue tracks that migration" half of #559's acceptance is satisfied.
